### PR TITLE
recombinate function

### PR DESCRIPTION
**Pull request by miguelarauj1o**
references #10, #19 and #18 (still working)

### CODE REVIEW

File: ackley/evolutionary_strategy/population.py [hunk 1]
```diff
@@ -110,3 +107,12 @@ def evolution(self, population):
...
         return [self.individual_type(self.args) for _ in range(0, population_size)]
+
+    def recombinateMean(self, parentOne, parentTwo):
+        # return
+
+    def recombinateRandom(self, parentOne, parentTwo):
+        if random.randint(0, 1):
+            return parentOne
+        else:
+            return parentTwo
```
Comment (JoaoGFarias) [inline ackley/evolutionary_strategy/population.py]: @miguelarauj1o You are returning one of the two parents. You should mix their chromosomes.
You can access the chromosomes using parentOne.chromosome['object_variable'] and parentTwo.chromosome['object_variable'].

Then you will have two lists of floats. You will create a new individual picking randomly floats from either list (You choose randomly for each item on the list)

File: ackley/evolutionary_strategy/population.py [hunk 2]
```diff
@@ -51,25 +64,23 @@ def generate_child(self, population):
...
         self.avaliations = 0
         return self.create_population(self.population_size)
 
-    @abc.abstractmethod
-    def create_population(self):
-        pass
+    def recombinate(self, parentOne, parentTwo):
+        x = parentOne.chromossome['object_variable']
+        y = parentTwo.chromossome['object_variable']
+        newChild = numpy.mean(x, y)
```
Comment (JoaoGFarias) [inline ackley/evolutionary_strategy/population.py]: Does it work?

screen shot 2016-05-24 at 2 16 51 am

Should'nt it be one list of lists?
screen shot 2016-05-24 at 2 17 32 am

The [docs](http://docs.scipy.org/doc/numpy-1.10.0/reference/generated/numpy.mean.html) say it just receive one array.


Anyway, it is taking the mean of all numbers. The algorithm said is mean index by index.

[1,2] and [3,5], e.g.

[ 1+3 / 2 , 2+5/2] => [2, 3.5]

File: ackley/evolutionary_strategy/population.py [hunk 3]
```diff
@@ -51,25 +64,23 @@ def generate_child(self, population):
...
+        return newChild
 
-class Uncorrelated_Algorithm(Evolutionary_Strategy):
+    def recombinateByRandomChoice(self, parentOne, parentTwo):
+        if random.randint(0, 1):
+            return parentOne.chromossome['object_variable']
+        else:
+            return parentTwo.chromossome['object_variable']
```
Comment (JoaoGFarias) [inline ackley/evolutionary_strategy/population.py]: It's returning the whole chromosome of the one of the two parents.
It should mix both chromosomes.

[1,2,3,4,5] and [6,7,8,9,10] can become [6, 2, 8, 4, 10]. 6 comes from the second, 2 from the first, 8 from the second, 4 from the first and 10 from the second.

File: ackley/evolutionary_strategy/population.py [hunk 4]
```diff
@@ -71,7 +71,11 @@ def start(self):
     def recombinate(self, parentOne, parentTwo):
         x = parentOne.chromossome['object_variable']
         y = parentTwo.chromossome['object_variable']
-        newChild = numpy.mean(x, y)
+        newChild = []
+
+        for (i, j) in zip(x, y):
```
Comment (miguelarauj1o) [inline ackley/evolutionary_strategy/population.py]: zip: http://stackoverflow.com/questions/1663807/how-can-i-iterate-through-two-lists-in-parallel-in-python
extend: http://stackoverflow.com/questions/252703/python-append-vs-extend

**Pull request closed without merging**